### ISSUE TITLE
Upserting row with timestamp without time zone results in time constraint violation

### ISSUE THREAD

**Issue by jessicaaustin** [timescale/timescaledb#313]
Versions: I've tried 0.4.2 and 0.6.1 (running docker image timescale/timescaledb-postgis)

Summary: Upserting row with timestamp without time zone results in time constraint violation

My guess is that somewhere along the way there is a timezone conversion happening, so that timescaledb uses one time when it figures out which chunk to insert the row, and another time when it actually inserts. (Similar to https://github.com/timescale/timescaledb/issues/306)

---

Table creation (this is run in a database called `sensors`):

```
CREATE TABLE device_feed_obs (
  fid   INTEGER          NOT NULL,
  time  TIMESTAMP        NOT NULL,
  value DOUBLE PRECISION NOT NULL,
  lat   NUMERIC(9, 6)    NULL,
  lon   NUMERIC(9, 6)    NULL,
  z     NUMERIC(6, 2)    NOT NULL DEFAULT 0,
  PRIMARY KEY (fid, time, z)
);

SELECT create_hypertable('device_feed_obs', 'time');
``` 

We're using a Kafka JDBC Sink Connector to insert data into this table. Recently we started getting errors that look like:

```
java.sql.BatchUpdateException: Batch entry 1 INSERT INTO "device_feed_obs" ("fid","time","z","value","lat","lon") VALUES (1,'2017-09-20T18:00:00',0.0,3.1999999999999886,NULL,NULL) ON CONFLICT ("fid","time","z") DO UPDATE SET "value"=EXCLUDED."value","lat"=EXCLUDED."lat","lon"=EXCLUDED."lon" was aborted.  Call getNextException to see the cause.
	at org.postgresql.jdbc2.AbstractJdbc2Statement$BatchResultHandler.handleError(AbstractJdbc2Statement.java:2778)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1912)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:338)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeBatch(AbstractJdbc2Statement.java:2959)
	at io.confluent.connect.jdbc.sink.BufferedRecords.flush(BufferedRecords.java:99)
	at io.confluent.connect.jdbc.sink.BufferedRecords.add(BufferedRecords.java:78)
	at io.confluent.connect.jdbc.sink.JdbcDbWriter.write(JdbcDbWriter.java:62)
	at io.confluent.connect.jdbc.sink.JdbcSinkTask.put(JdbcSinkTask.java:66)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:429)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:250)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:179)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:148)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:139)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

(If it's relevant, in this case this row already exists and the `value` is getting updated.)

On the database side the error looks like:

```
ERROR:  new row for relation "_hyper_1_1_chunk" violates check constraint "constraint_1"
DETAIL:  Failing row contains (1, 2017-09-20 18:00:00, 3.19999999999998863, null, null, 0.00).
STATEMENT:  INSERT INTO "device_feed_obs" ("fid","time","z","value","lat","lon") VALUES ($1,$2,$3,$4,$5,$6) ON CONFLICT ("fid","time","z") DO UPDATE SET "value"=EXCLUDED."value","lat"=EXCLUDED."lat","lon"=EXCLUDED."lon"
```

Looking at `_timescaledb_internal`:

```
sensors=# \d _timescaledb_internal.*;

... truncated ...

      Table "_timescaledb_internal._hyper_1_1_chunk"
 Column |            Type             |     Modifiers      
--------+-----------------------------+--------------------
 fid    | integer                     | not null
 time   | timestamp without time zone | not null
 value  | double precision            | not null
 lat    | numeric(9,6)                | 
 lon    | numeric(9,6)                | 
 z      | numeric(6,2)                | not null default 0
Indexes:
    "1_1_device_feed_obs_pkey" PRIMARY KEY, btree (fid, "time", z)
    "_hyper_1_1_chunk_device_feed_obs_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_1" CHECK ("time" >= '2017-08-21 16:00:00'::timestamp without time zone AND "time" < '2017-09-20 16:00:00'::timestamp without time zone)
Inherits: device_feed_obs

Index "_timescaledb_internal._hyper_1_1_chunk_device_feed_obs_time_idx"
 Column |            Type             | Definition 
--------+-----------------------------+------------
 time   | timestamp without time zone | "time"
btree, for table "_timescaledb_internal._hyper_1_1_chunk"

      Table "_timescaledb_internal._hyper_1_2_chunk"
 Column |            Type             |     Modifiers      
--------+-----------------------------+--------------------
 fid    | integer                     | not null
 time   | timestamp without time zone | not null
 value  | double precision            | not null
 lat    | numeric(9,6)                | 
 lon    | numeric(9,6)                | 
 z      | numeric(6,2)                | not null default 0
Indexes:
    "2_2_device_feed_obs_pkey" PRIMARY KEY, btree (fid, "time", z)
    "_hyper_1_2_chunk_device_feed_obs_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_2" CHECK ("time" >= '2017-06-22 16:00:00'::timestamp without time zone AND "time" < '2017-07-22 16:00:00'::timestamp without time zone)
Inherits: device_feed_obs

Index "_timescaledb_internal._hyper_1_2_chunk_device_feed_obs_time_idx"
 Column |            Type             | Definition 
--------+-----------------------------+------------
 time   | timestamp without time zone | "time"
btree, for table "_timescaledb_internal._hyper_1_2_chunk"

      Table "_timescaledb_internal._hyper_1_3_chunk"
 Column |            Type             |     Modifiers      
--------+-----------------------------+--------------------
 fid    | integer                     | not null
 time   | timestamp without time zone | not null
 value  | double precision            | not null
 lat    | numeric(9,6)                | 
 lon    | numeric(9,6)                | 
 z      | numeric(6,2)                | not null default 0
Indexes:
    "3_3_device_feed_obs_pkey" PRIMARY KEY, btree (fid, "time", z)
    "_hyper_1_3_chunk_device_feed_obs_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_3" CHECK ("time" >= '2017-09-20 16:00:00'::timestamp without time zone AND "time" < '2017-10-20 16:00:00'::timestamp without time zone)
Inherits: device_feed_obs

Index "_timescaledb_internal._hyper_1_3_chunk_device_feed_obs_time_idx"
 Column |            Type             | Definition 
--------+-----------------------------+------------
 time   | timestamp without time zone | "time"
btree, for table "_timescaledb_internal._hyper_1_3_chunk"

      Table "_timescaledb_internal._hyper_1_4_chunk"
 Column |            Type             |     Modifiers      
--------+-----------------------------+--------------------
 fid    | integer                     | not null
 time   | timestamp without time zone | not null
 value  | double precision            | not null
 lat    | numeric(9,6)                | 
 lon    | numeric(9,6)                | 
 z      | numeric(6,2)                | not null default 0
Indexes:
    "4_4_device_feed_obs_pkey" PRIMARY KEY, btree (fid, "time", z)
    "_hyper_1_4_chunk_device_feed_obs_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_4" CHECK ("time" >= '2017-10-20 16:00:00'::timestamp without time zone AND "time" < '2017-11-19 15:00:00'::timestamp without time zone)
Inherits: device_feed_obs

Index "_timescaledb_internal._hyper_1_4_chunk_device_feed_obs_time_idx"
 Column |            Type             | Definition 
--------+-----------------------------+------------
 time   | timestamp without time zone | "time"
btree, for table "_timescaledb_internal._hyper_1_4_chunk"

```

So it's trying to insert a row into `_hyper_1_1_chunk` with `time='2017-09-20 18:00:00'` which is obviously going to fail the constraint with `"time" < '2017-09-20 16:00:00'`. It should go into `_hyper_1_3_chunk` right?

I can also run the insert from psql and get the same error:

```
sensors=# INSERT INTO "device_feed_obs" ("fid","time","z","value","lat","lon") VALUES (1,'2017-09-20T18:00:00',0.0,3.1999999999999886,NULL,NULL) ON CONFLICT ("fid","time","z") DO UPDATE SET "value"=EXCLUDED."value","lat"=EXCLUDED."lat","lon"=EXCLUDED."lon";
ERROR:  new row for relation "_hyper_1_1_chunk" violates check constraint "constraint_1"
DETAIL:  Failing row contains (1, 2017-09-20 18:00:00, 3.19999999999999, null, null, 0.00).
```

**Comment by jessicaaustin** [comment]
Somehow I missed this issue: https://github.com/timescale/timescaledb/issues/300 

Looks like the same problem I'm seeing. I'll test with the patch and close this if it resolves the issue.

**Comment by jessicaaustin** [comment]
Issue has been resolved in 0.7.0
